### PR TITLE
Fd 2777 update design system for special cards

### PR DESCRIPTION
**Pull request by AlexanderEerenberg**

### CODE REVIEW

**Comment by willemliufdmg** [comment]
#### 📝 Changed routes:


| [**/card/cards.module**](https://fd-design-system-1to2sjisf-fdmediagroep.vercel.app/card/cards.module) | [**/card/horizontal-1**](https://fd-design-system-1to2sjisf-fdmediagroep.vercel.app/card/horizontal-1) |
|:-:|:-:|
|<a href="https://fd-design-system-1to2sjisf-fdmediagroep.vercel.app/card/cards.module">Screenshot of /card/cards.module</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/Paxb6KYukkAIE5REi_BQFUPOVeJ9AT7_tJ_txWZ1OstC66nVW3dkEsxonTbyyXhRXzlAnjYLdC-3OBeI7gNJoA.png">(view full size)</a> |<a href="https://fd-design-system-1to2sjisf-fdmediagroep.vercel.app/card/horizontal-1">Screenshot of /card/horizontal-1</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/XDyAbj6HFrLJlpaaEsrbPNMQKJeog2uk5HT9pcjiwtEZ7man-E0eF7phEsmK0ObuGr0gWbH1IQJwfeZk8qLteg.png">(view full size)</a> |



| [**/card/horizontal-4**](https://fd-design-system-1to2sjisf-fdmediagroep.vercel.app/card/horizontal-4) | [**/card/hybrid-4**](https://fd-design-system-1to2sjisf-fdmediagroep.vercel.app/card/hybrid-4) |
|:-:|:-:|
|<a href="https://fd-design-system-1to2sjisf-fdmediagroep.vercel.app/card/horizontal-4">Screenshot of /card/horizontal-4</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/K1SoIJYkpzJPCWrO1pwtHJdcz8bloAdqqM9lc4GC-Fl0uvj_52Ubcm3buGGr7Ti1aD_qP4Y1e7O733ayful_SQ.png">(view full size)</a> |<a href="https://fd-design-system-1to2sjisf-fdmediagroep.vercel.app/card/hybrid-4">Screenshot of /card/hybrid-4</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/Oe9yL426flibIxWs4l3_rysiBMnt4n4Ng3Jlz6oQuYh6G4vxtZIZCqK0dIwZTZcG5X0XY0P6keFMUXHnFr-uAQ.png">(view full size)</a> |



| [**/card/vertical-6**](https://fd-design-system-1to2sjisf-fdmediagroep.vercel.app/card/vertical-6) | [**/cards**](https://fd-design-system-1to2sjisf-fdmediagroep.vercel.app/cards) |
|:-:|:-:|
|<a href="https://fd-design-system-1to2sjisf-fdmediagroep.vercel.app/card/vertical-6">Screenshot of /card/vertical-6</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/QPCbTW7QD2BpyJ8qrhyil6GXkNg8oK2QXcgVfL2LMXG8hEgI-z8DBVKHP5QtYRZMHeXc3rdHP-8OWFXlWfijTQ.png">(view full size)</a> |<a href="https://fd-design-system-1to2sjisf-fdmediagroep.vercel.app/cards">Screenshot of /cards</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/M4-6sgQuHqPEdLRGRqIpCqeIBxq-piU18z1bWWToPLQ7R6FIOMSqlPrbVK6kFtlJ4uF6X3mit_ZpV7jldROuaQ.png">(view full size)</a> |







Commit 952bcadb76905dde444f460cc70443c8b3a402c0 (https://fd-design-system-1to2sjisf-fdmediagroep.vercel.app).

File: src/components/card/HybridCard4.tsx [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    url: string;
+    label: string;
+    imageComponent?: JSX.Element;
+    caption?: string;
+    title: string;
+    [x: string]: any;
+}
+
+/**
+ * Used as breaking.
```
Comment (willemliufdmg) [inline src/components/card/HybridCard4.tsx]: Is it used as breaking?

File: src/components/card/HybridCard2.tsx [hunk 2]
```diff
@@ -6,10 +6,7 @@ export type Variants = 'default' | 'variant-1';
 interface Props {
     id: string;
     url: string;
-    imageUrl?: string;
-    imageUrlS?: string;
-    imageUrlM?: string;
-    imageUrlL?: string;
+    imageComponent?: JSX.Element;
```
Comment (willemliufdmg) [inline src/components/card/HybridCard2.tsx]: What happened here? I don't see imageComponent being used in this file. But there are still references to `props.imageUrlS`, `props.imageUrlM`, `props.imageUrlL`.

File: src/components/article-link-card/LinkCard.tsx [hunk 3]
```diff
@@ -22,7 +22,7 @@ export default function LinkCard(props: Props) {
                     {props.linkText
                         ? props.linkText
                         : 'Beheer uw nieuwsbriefinstellingen'}{' '}
-                    <span
+                    <div
```
Comment (willemliufdmg) [inline src/components/article-link-card/LinkCard.tsx]: Can be left as `span`?

File: src/components/card/HorizontalCard4.module.scss [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+    box-shadow: 0 2px 0.2px rgba(0, 0, 0, 0.1);
+    position: relative;
+
+    h1 {
+        color: #{map-deep-get($colors, 'colors', 'neutral', 90)};
+        // IE11
+        @media screen and (-ms-high-contrast: active),
+            (-ms-high-contrast: none) {
+            color: #{map-deep-get($colors, 'colors', 'neutral', 10)};
+        }
```
Comment (willemliufdmg) [inline src/components/card/HorizontalCard4.module.scss]: Can remove IE11 support

```suggestion
```

File: src/components/card/HorizontalCard4.module.scss [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+.fdHorizontalCard4 {
+    transition: box-shadow 0.15s linear;
+    position: relative;
+    margin-bottom: 1px;
+    justify-content: space-between;
+    background: #{map-deep-get($colors, 'colors', 'primary', 75)};
+    // IE11
+    @media screen and (-ms-high-contrast: active), (-ms-high-contrast: none) {
+        background: #{map-deep-get($colors, 'colors', 'neutral', 90)};
+    }
```
Comment (willemliufdmg) [inline src/components/card/HorizontalCard4.module.scss]: Can remove IE11 support

```suggestion
```

File: src/components/card/HybridCard2.tsx (hunk 2 above)
Comment (willemliufdmg) [reply src/components/card/HybridCard2.tsx]: The model in the example page contains the imageUrlS etc. urls:

image

That's why it seems to render correctly.

But the component interface doesn't describe these props anymore and expects an imageComponent which isn't used in the component itself. Maybe the interface needs to be reverted to what it was?

Comment (AlexanderEerenberg) [reply src/components/card/HybridCard2.tsx]: I will remove the imageComponent prop. We should really get rid of [x: string]: any; as it makes it confusing which props are still used.

File: src/components/article-link-card/LinkCard.tsx (hunk 3 above)
Comment (AlexanderEerenberg) [reply src/components/article-link-card/LinkCard.tsx]: Fixed this one earlier

File: src/components/card/HybridCard2.tsx (hunk 2 above)
Comment (willemliufdmg) [reply src/components/card/HybridCard2.tsx]: The old imageUrlS etc. props need to be re-added as well

**Review by willemliufdmg — APPROVED** [review]

**Pull request merged**